dataService.js

```diff
@@ -0,0 +1,91 @@
+// Maps.
+let productMap;
+let categoryMap;
+let combinedProductMap;
+
+// Assign product ID as key and all other product information as value in map.
+function getProducts() {
+
+  // If productMap is not empty, return productMap.
+  if (productMap) {
+    return productMap;
+  } else {
+
+    // Otherwise, load the JSONs and create the map.
+    const Product = require('./src/data/products.json');
+    const productMapForReturn = new Map();
+
+    for (let food of Product.products) {
+      productMapForReturn.set(food.id, food);
+    };
+
+    productMap = productMapForReturn;
+    return productMap;
+
+  }
+
+}
+
+// Assign category ID as key and category name as value in map.
+function getCategories() {
+
+  // If categoryMap is not empty, return categoryMap.
+  if (categoryMap) {
+    return categoryMap;
+  } else {
+
+    // Otherwise, load the JSONs and create the map.
+    const Category = require('./src/data/categories.json');
+    const categoryMapForReturn = new Map();
+
+    Category.categories.forEach(category => {
+      categoryMapForReturn.set(category.id, category);
+    });
+
+    categoryMap = categoryMapForReturn;
+    return categoryMap;
+
+  }
+
+}
+
+// Assign products to the appropriate categories.
+function combineProductsWithCategories() {
+
+  const Product = require('./src/data/products.json');
+  const Category = require('./src/data/categories.json');
```

Comment:
See above comment. Also, you should only use `productMap` and `categoryMap` in this function, no raw data involved.

Reply:
OK, fixed.